tests/check_new_syntax.py

```diff
@@ -62,31 +92,51 @@ def visit_Subscript(self, node: ast.Subscript) -> None:
     # TODO: can use built-in generics in type aliases
     class AnnotationFinder(ast.NodeVisitor):
         def __init__(self) -> None:
-            self.set_from_collections_abc = False
+            self.module_imports_info_dict = ModuleImportsDict(
+                set_from_collections_abc=False, context_manager_from_typing=False, async_context_manager_from_typing=False
+            )
+
+        def old_syntax_finder(self) -> OldSyntaxFinder:
+            """Convenience method to create an `OldSyntaxFinder` method with the correct state"""
+            return OldSyntaxFinder(self.module_imports_info_dict)
 
         def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
             if node.module == "collections.abc":
-                imported_classes = node.names
-                if any(cls.name == "Set" for cls in imported_classes):
-                    self.set_from_collections_abc = True
+                classes_from_collections_abc = node.names
+                if any(cls.name == "Set" for cls in classes_from_collections_abc):
+                    self.module_imports_info_dict["set_from_collections_abc"] = True
+
+            elif node.module == "typing":
+                classes_from_typing = node.names
+
+                for cls in classes_from_typing:
+                    cls_name = cls.name
+                    if cls_name == "ContextManager":
+                        self.module_imports_info_dict["context_manager_from_typing"] = True
+                        if self.module_imports_info_dict["async_context_manager_from_typing"]:
+                            break
+                    elif cls_name == "AsyncContextManager":
+                        self.module_imports_info_dict["async_context_manager_from_typing"] = True
+                        if self.module_imports_info_dict["context_manager_from_typing"]:
+                            break
```

Comment:
Why do we need to stop the loop when both are found?

Reply:
Just a micro-optimisation, doesn't really need to be there, happy to take it out